catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/model/Association.js

```diff
@@ -142,15 +143,17 @@ define([
         },
         removeSegment: function (segment) {
             var model = this;
-            var updatedAssociations = [];
+            var removedAssociations = [];
             var associations = this.get('associations').models;
             var segId = segment.get('segmentId');
             _.each(associations, function (association) {
-                if (association.get('sourceId') !== segId && association.get('targetId') !== segId) {
-                    updatedAssociations.push(association);
+                if (association.get('sourceId') === segId || association.get('targetId') === segId) {
+                    removedAssociations.push(association);
                 }
             });
-            this.set('associations', updatedAssociations);
+            _.each(removedAssociations, function(association){
```

Comment:
Why loop through this again? Couldn't you just do `model.get('associations').remove(association);` in the above loop and get rid of the array entirely?